docs/configuration/indexes.md

```diff
@@ -175,27 +175,47 @@ url = "https://public:koala@pypi-proxy.corp.dev/simple"
 For security purposes, credentials are _never_ stored in the `uv.lock` file; as such, uv _must_ have
 access to the authenticated URL at installation time.
 
-## Configuring authentication for indexes
+## Using credential providers
```

Comment:
I'm a little hesitant about this framing, but I think it matches the other sections better and will be more discoverable.

We'll link to this from the index-specific documentation that requires this feature.